### PR TITLE
adds qgis-plugin-ci to plugins tips and tricks

### PR DESCRIPTION
**Pull request by haubourg**
@Gustry @m-kuhn @3nids  Following the qgis-plugin-ci nice presentation today, I propose to add it to the documentation tips and tricks next to the minimal plugin and qgis3 builder. 
Does it look goo to you?

### CODE REVIEW

File: docs/pyqgis_developer_cookbook/plugins/plugins.rst [hunk 1]
```diff
@@ -69,14 +69,16 @@ What is the meaning of the files:
...
 is a way of creating the basic files (skeleton) of a typical
 QGIS Python plugin.
 
 There is a QGIS plugin called
 `Plugin Builder 3 <https://plugins.qgis.org/plugins/pluginbuilder3/>`_
 that creates a plugin template for QGIS.
-This is the recommended option, as it produces 3.x compatible sources.
+It produces 3.x compatible sources, with full possible features for a plugin.
+
+Another productivity option is using `qgis-plugin-ci <https://opengisch.github.io/qgis-plugin-ci/>`_ CLI to perform automated testing and deployment for QGIS plugins using continuous integration on Travis-CI, github workflows, Gitlab-CI and Transifex for translation. It allows to release, translate, publish or create a XML plugin repository file via CLI or automated CI actions.
```
Comment (m-kuhn) [inline docs/pyqgis_developer_cookbook/plugins/plugins.rst:81]:
```suggestion
Another productivity option is using `qgis-plugin-ci CLI to perform automated testing and deployment for QGIS plugins<https://opengisch.github.io/qgis-plugin-ci/>`_ using continuous integration on `GitHub workflows <https://docs.github.com/en/actions/using-workflows>`_, `Gitlab-CI <https://docs.gitlab.com/ee/ci/>`_ or other ci systems. It integrates `Transifex <https://www.transifex.com/>`_ for translation, allows releasing, and publishing to the plugin repository as well as creating a private plugin repository via CLI or automated CI actions. 
```

File: docs/pyqgis_developer_cookbook/plugins/plugins.rst [hunk 2]
```diff
@@ -69,14 +69,16 @@ What is the meaning of the files:
 * :file:`metadata.txt` = Contains general info, version, name and some other
   metadata used by plugins website and plugin infrastructure.
 
-`Here <https://github.com/wonder-sk/qgis-minimal-plugin>`_
+For training purpose `Here <https://github.com/wonder-sk/qgis-minimal-plugin>`_
```
Comment (m-kuhn) [inline docs/pyqgis_developer_cookbook/plugins/plugins.rst:72]: I know it's not your phrasing, but could we avoid using "here" as link and instead use something like "a minimal plugin template to get started" or similar as clickable? (I cannot make suggestions over lines that are not included in your diff, sorry)

File: docs/pyqgis_developer_cookbook/plugins/plugins.rst (hunk 1 above)
Comment (Gustry) [inline docs/pyqgis_developer_cookbook/plugins/plugins.rst:81]:
```suggestion
Another productivity option is using `qgis-plugin-ci <https://opengisch.github.io/qgis-plugin-ci/>`_ CLI to perform automated packaging and deployment for QGIS plugins using continuous integration on Travis-CI, github workflows, Gitlab-CI and Transifex for translation. It allows to release, translate, publish or create a XML plugin repository file via CLI or automated CI actions. 
```

The tool doesn't not provide a testing environment, for now.

Can you check the wordwrap in your editor as well ? CF your changes https://raw.githubusercontent.com/qgis/QGIS-Documentation/67f4841ab9687648a2e397ea1471d5fc439d4e9c/docs/pyqgis_developer_cookbook/plugins/plugins.rst

File: docs/pyqgis_developer_cookbook/plugins/plugins.rst (hunk 2 above)
Comment (haubourg) [reply docs/pyqgis_developer_cookbook/plugins/plugins.rst]: Totally agreed.I reworded this. 
I also moved qgis-ci plugin into the tips and trip section, where it belongs more than here where we describe how to initialize a new plugin

File: docs/pyqgis_developer_cookbook/plugins/plugins.rst [hunk 3]
```diff
@@ -88,6 +78,15 @@ Another productivity option is using `qgis-plugin-ci <https://opengisch.github.i
 
 .. index:: Plugins; Writing code
 
+Useful tools to quickly create plugins
+--------------------------------------
+
+`A minimal plugin to get started <https://github.com/wonder-sk/qgis-minimal-plugin>`_
+that only creates the basic files (skeleton) of a typical QGIS Python plugin.
```
Comment (m-kuhn) [inline docs/pyqgis_developer_cookbook/plugins/plugins.rst:85]:
```suggestion
that only includes the basic files (skeleton) of a typical QGIS Python plugin.
```

File: docs/pyqgis_developer_cookbook/plugins/plugins.rst [hunk 4]
```diff
@@ -88,6 +78,15 @@ Another productivity option is using `qgis-plugin-ci <https://opengisch.github.i
...
+
+A fully featured QGIS plugin called `Plugin Builder 3 <https://plugins.qgis.org/plugins/pluginbuilder3/>`_
+that creates a plugin template for QGIS. It produces 3.x compatible sources, with full possible features for a plugin.
```
Comment (m-kuhn) [inline docs/pyqgis_developer_cookbook/plugins/plugins.rst:88]:
```suggestion
that creates a plugin template for QGIS. It produces 3.x compatible sources, full of possible features for a plugin.
```

?

Comment (m-kuhn) [reply docs/pyqgis_developer_cookbook/plugins/plugins.rst:88]: or

```suggestion
that creates a plugin template for QGIS. It produces 3.x compatible sources, with many possible features for a plugin.
```

Comment (haubourg) [reply docs/pyqgis_developer_cookbook/plugins/plugins.rst:88]: option 2 for me

File: docs/pyqgis_developer_cookbook/plugins/plugins.rst [hunk 5]
```diff
@@ -598,6 +599,13 @@ During development of your plugin you will frequently need to reload it in QGIS
 for testing. This is very easy using the **Plugin Reloader** plugin. You can find it
 with the :ref:`Plugin Manager <plugins>`.
 
+Automate packaging, release and translation with qgis-plugin-ci
+---------------------------------------------------------------
+
+`qgis-plugin-ci <https://opengisch.github.io/qgis-plugin-ci/>`_ provides a command line interface to perform automated packaging and deployment for QGIS plugins on your laptop, or using continuous integration on Travis-CI, github workflows, Gitlab-CI and Transifex for translation.
```
Comment (m-kuhn) [inline docs/pyqgis_developer_cookbook/plugins/plugins.rst:605]:
```suggestion
`qgis-plugin-ci <https://opengisch.github.io/qgis-plugin-ci/>`_ provides a command line interface to perform automated packaging and deployment for QGIS plugins on your computer, or using continuous integration 
like `GitHub workflows <https://docs.github.com/en/actions/using-workflows>`_ or `Gitlab-CI <https://docs.gitlab.com/ee/ci/>`_ as well as `Transifex <https://www.transifex.com/>`_ for translation.
```

- Removed travis (is that still used?)
- Added some links

File: docs/pyqgis_developer_cookbook/plugins/plugins.rst [hunk 6]
```diff
@@ -598,6 +599,13 @@ During development of your plugin you will frequently need to reload it in QGIS
...
+
+It allows to release, translate, publish or even a XML plugin repository file via CLI or automate this in CI actions.
```
Comment (m-kuhn) [inline docs/pyqgis_developer_cookbook/plugins/plugins.rst:607]:
```suggestion
It allows releasing, translating, publishing or generating an XML plugin repository file via CLI or in CI actions. 
```

Some readability improvements

**Comment by haubourg** [comment]
@DelazJ looks good to you ?

**Review by m-kuhn — APPROVED** [review]

**Comment by haubourg** [comment]
Merci @DelazJ !

**Comment by DelazJ** [comment]
👍

**Pull request merged**